### PR TITLE
Rewrite of na_ontap_snapshot to follow NetAppModule, add new option to rename snapshot, bug fix to modify snapmirror_label

### PR DESCRIPTION
**Pull request by carchi8py**
##### SUMMARY
Rewrite to fit NetAppModule
Add new option from_name to rename snapshot 
bug fix to allow modifying of snapmirror_label

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
- na_ontap_snapshot

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test/units/modules/storage/netapp/test_na_ontap_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+from units.compat.mock import patch
+from ansible.module_utils import basic
+from ansible.module_utils._text import to_bytes
+import ansible.module_utils.netapp as netapp_utils
+
+from ansible.modules.storage.netapp.na_ontap_snapshot \
+    import NetAppOntapSnapshot as my_module
+
+if not netapp_utils.has_netapp_lib():
+    pytestmark = pytest.skip('skipping as missing required netapp_lib')
```
Comment (lonico) [inline test/units/modules/storage/netapp/test_na_ontap_snapshot.py]: pytest.mark.skip

How come this one is not up to date?

**Review by lonico — CHANGES_REQUESTED** [review]
pytest.mark.skip

**Review by lonico — APPROVED** [review]
shipit

**Review by thedoubl3j — APPROVED** [review]
rewrite looks good, no big changes aside from qol changes and `module_utils`. Tests added always good. lgtm

**Pull request merged**